drivers/sensor/tmag5273/tmag5273.c

```diff
@@ -0,0 +1,1161 @@
+
+	if (retval != 0) {
+		return retval;
+	}
+
+	regdata &= (TMAG5273_X_Y_RANGE_MSK | TMAG5273_Z_RANGE_MSK);
+
+	if (regdata == TMAG5273_XYZ_MEAS_RANGE_HIGH) {
+		val->val1 = (drv_data->version == TMAG5273_VER_TMAG5273X1)
+				    ? TMAG5273_MEAS_RANGE_HIGH_MT_VER1
+				    : TMAG5273_MEAS_RANGE_HIGH_MT_VER2;
+	} else {
+		val->val1 = (drv_data->version == TMAG5273_VER_TMAG5273X1)
+				    ? TMAG5273_MEAS_RANGE_LOW_MT_VER1
+				    : TMAG5273_MEAS_RANGE_LOW_MT_VER2;
+	}
+
+	return 0;
+}
+
+/**
+ * set the X/Y/Z angle & magnitude calculation mode
+ *
+ * @param dev handle to the sensor
+ * @param val value to be set
+ *
+ * @return \c -ENOTSUP if unknown value
+ * @return see @ref i2c_reg_update_byte_dt
+ */
+static inline int tmag5273_attr_set_xyz_calc(const struct device *dev,
+					     const struct sensor_value *val)
+{
+	const struct tmag5273_config *drv_cfg = dev->config;
+	uint8_t regval = 0;
+	int retval = -ENOTSUP;
+
+	switch (val->val1) {
+	case TMAG5273_ANGLE_CALC_NONE:
+		regval = TMAG5273_ANGLE_EN_NONE;
+		retval = 0;
+		break;
+	case TMAG5273_ANGLE_CALC_XY:
+		if (drv_cfg->x_axis && drv_cfg->y_axis) {
+			regval = TMAG5273_ANGLE_EN_XY;
+			retval = 0;
+		}
+		break;
+	case TMAG5273_ANGLE_CALC_YZ:
+		if (drv_cfg->y_axis && drv_cfg->z_axis) {
+			regval = TMAG5273_ANGLE_EN_YZ;
+			retval = 0;
+		}
+		break;
+	case TMAG5273_ANGLE_CALC_XZ:
+		if (drv_cfg->x_axis && drv_cfg->z_axis) {
+			regval = TMAG5273_ANGLE_EN_XZ;
+			retval = 0;
+		}
+		break;
+	default:
+		break;
+	}
+
+	if (retval == 0) {
+		retval = i2c_reg_update_byte_dt(&drv_cfg->i2c, TMAG5273_REG_SENSOR_CONFIG_2,
+						TMAG5273_ANGLE_EN_MSK,
+						regval << TMAG5273_ANGLE_EN_POS);
+	} else {
+		LOG_ERR("unknown attribute value |%d|", val->val1);
+	}
+
+	return retval;
+}
+
+/**
+ * returns the X/Y/Z angle & magnitude calculation mode
+ *
+ * @param dev handle to the sensor
+ * @param val return value
+ *
+ * @return \c 0 on success
+ * @return see @ref i2c_reg_read_byte_dt
+ */
+static inline int tmag5273_attr_get_xyz_calc(const struct device *dev, struct sensor_value *val)
+{
+	const struct tmag5273_config *drv_cfg = dev->config;
+
+	uint8_t regdata = 0;
+	int retval = i2c_reg_read_byte_dt(&drv_cfg->i2c, TMAG5273_REG_SENSOR_CONFIG_2, &regdata);
+
+	if (retval != 0) {
+		return retval;
+	}
+
+	val->val1 = (regdata & TMAG5273_ANGLE_EN_MSK) >> TMAG5273_ANGLE_EN_POS;
+
+	return 0;
+}
+
+static int tmag5273_attr_set(const struct device *dev, enum sensor_channel chan,
+			     enum sensor_attribute attr, const struct sensor_value *val)
+{
+	CHECKIF(dev == NULL) {
+		LOG_ERR("dev: NULL");
+		return -EINVAL;
+	}
+
+	CHECKIF(val == NULL) {
+		LOG_ERR("val: NULL");
+		return -EINVAL;
+	}
+
+	int retval = 0;
+
+	if (chan != SENSOR_CHAN_MAGN_XYZ) {
+		return -ENOTSUP;
+	}
+
+	const struct tmag5273_config *drv_cfg = dev->config;
+
+	retval = -ENOTSUP;
+
+	switch ((uint16_t)attr) {
+	case TMAG5273_MEAS_RANGE:
```

Comment:
This should probably be using the common SENSOR_ATTR_FULL_RANGE for setting the measurement range